mailserver/mailserver.go

```diff
@@ -53,9 +55,17 @@ var (
 	archivedErrorsCounter  = metrics.NewRegisteredCounter("mailserver/archiveErrors", nil)
 )
 
+type DB interface {
```

Comment:
I would add a comment explaining that this interface is purely for testing a different leveldb implementation, i.e. `panicDB`. Maybe even make it private if possible?

A proper interface, in this case, would not have any references to `leveldb` types.

Reply:
yes you are right, maybe something like `dbImpl`?